crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -122,14 +123,24 @@ pub trait LoadPendingBlock: EthApiTypes {
     /// Returns the locally built pending block
     fn local_pending_block(
         &self,
-    ) -> impl Future<Output = Result<Option<SealedBlockWithSenders>, Self::Error>> + Send
+    ) -> impl Future<Output = Result<Option<(SealedBlockWithSenders, Vec<Receipt>)>, Self::Error>> + Send
     where
         Self: SpawnBlocking,
     {
         async move {
             let pending = self.pending_block_env_and_cfg()?;
             if pending.origin.is_actual_pending() {
-                return Ok(pending.origin.into_actual_pending())
+                if let Some(block) = pending.origin.into_actual_pending() {
+                    let receipts = self
+                        .provider()
+                        .receipts_by_block(BlockHashOrNumber::Number(block.number))
+                        .map_err(Self::Error::from_eth_err)?
+                        .ok_or_else(|| {
+                            Self::Error::from_eth_err(EthApiError::UnknownBlockNumber)
```

Comment:
another error would be more suitable here, since this only gets called in case `pending.origin.into_actual_pending()` actually returns `Some(block)`